### PR TITLE
Flask: finish implementing progressive loading and filtering on /api/participants

### PR DESCRIPTION
**Pull request by kevinlul**
Update default data to provide multiple children in one-to-many relationships to help test .count() behaviour.
Fix a severe N+1 type problem in this endpoint and potential security problems with user input feeding into `getattr`

#283

### CODE REVIEW

**Comment by kevinlul** [comment]
I messed up escaping user input to LIKE. We need to make sure that when users search for `%` and similar, it's searching for a literal `%` and not interpreted.

File: flask/app/participants.py [hunk 1]
```diff
@@ -33,75 +38,101 @@
...
+                models.ParticipantType,
+                participant_type,
+            )
+        )
+    sex = request.args.get("sex", type=str)
+    if sex:
+        filters.append(filter_in_enum_or_abort(models.Participant.sex, models.Sex, sex))
+    affected = request.args.get("affected", type=str)
+    if affected:
+        filters.append(filter_nullable_bool_or_abort(models.Participant, affected))
```
Comment (Madelinehazel) [inline flask/app/participants.py:115]:
```suggestion
        filters.append(filter_nullable_bool_or_abort(models.Participant.affected, affected))
```

File: flask/app/participants.py [hunk 2]
```diff
@@ -33,75 +38,101 @@
...
+    solved = request.args.get("solved", type=str)
+    if solved:
+        filters.append(filter_nullable_bool_or_abort(models.Participant, solved))
```
Comment (Madelinehazel) [inline flask/app/participants.py:118]:
```suggestion
        filters.append(filter_nullable_bool_or_abort(models.Participant.solved, solved))
```

File: flask/app/participants.py [hunk 3]
```diff
@@ -33,75 +38,101 @@
...
+    if family_codename:
+        filters.append(func.instr(models.Family.family_codename, family_codename))
+    participant_codename = request.args.get("participant_codename", type=str)
+    if participant_codename:
+        filters.append(
+            func.instr(models.Participant.participant_codename, participant_codename)
+        )
+    notes = request.args.get("notes", type=str)
+    if notes:
+        filters.append(func.instr(models.Participant.notes, notes))
```
Comment (Madelinehazel) [inline flask/app/participants.py:100]: We also need to be able to handle 'notes=null'

**Review by Madelinehazel — CHANGES_REQUESTED** [review]
Nice work!

File: flask/app/participants.py (hunk 3 above)
Comment (kevinlul) [reply flask/app/participants.py:100]: What would that do?

**Review by frewmack — COMMENTED** [review]
Frontend works as expected after slight changes to accommodate "{column}={value}" format. Should I push the changes here instead of a separate PR?

**Review by delvinso — APPROVED** [review]
Great work, this looks like it was a huge undertaking! Looks good to me.

**Comment by kevinlul** [comment]
Separate PR is fine so this doesn't get more messy than it needs to

**Review by frewmack — APPROVED** [review]

File: flask/app/participants.py (hunk 3 above)
Comment (Madelinehazel) [reply flask/app/participants.py:100]: Get records where there are no notes associated with the participant. Right now, if you pass notes=null, no participants are returned. BUT, I guess this isn't relevant from a use-case point of view (presumably, we wouldn't need to filter for participants that do not have notes associated with them).

**Pull request merged**